pkg/kv/bulk/sst_batcher.go

```diff
@@ -692,7 +692,17 @@ func (b *SSTBatcher) doFlush(ctx context.Context, reason int) error {
 	// the next one; if memory is not available we'll just block on the send
 	// and then move on to the next send after this SST is no longer being held
 	// in memory.
-	flushAsync := reason == rangeFlush
+	//
+	// TODO(jeffswenson): re-enable flush async after fixing performance and
+	// correctness issues.
+	//
+	// CORRECTNESS: Something has to surface the error from the async flush to
```

Comment:
for the correctness point, https://github.com/cockroachdb/cockroach/pull/144595 is ready for review

Reply:
I think the lack of error handling in reset is actually okay, but we need to remove the wait and remove reset from the public API. The `Flush()` + `Close()` API works well where `Flush()` is the blocking API that returns errors to the users and `Close()` performs clean up in the unwind case. 

The reason https://github.com/cockroachdb/cockroach/pull/110218 was merged is it was using `Reset` in a place where we should have been closing the old batcher and creating a new batcher.